lib/ansible/inventory/manager.py

```diff
@@ -129,8 +130,7 @@ def split_host_pattern(pattern):
                     )+              # occurring once or more
                 '''), pattern, re.X
             )
-
-    return [p.strip() for p in patterns]
+    return [p for p in patterns if p]
```

Comment:
patterns from the regex won't be stripped?

also this traverses the list x2, not sure what is more expensive, strip x2 or list x2

Reply:
Doh.

I went back to doing the "double strip" with a comprehension. With 100,000 hosts a regular `for` loop performed the same as doing the comprehension with the "double strip" on both Python 2 and 3.

```
patterns = [' host{} '.format(n) for n in range(1,100000)]

# Python 3
In [29]: %timeit split_host_pattern_loop(patterns)
1.06 s ± 66.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

In [30]: %timeit split_host_pattern_ds(patterns)
1.01 s ± 17.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


# Python 2
In [14]: %timeit split_host_pattern_loop(patterns)
1 loop, best of 3: 1.36 s per loop

In [18]: %timeit split_host_pattern_ds(patterns)
1 loop, best of 3: 1.37 s per loop
```

Reply:
you REAAAALLY don't want a `for ` loop ....